src/tests/baseservices/compilerservices/UnsafeAccessors/UnsafeAccessorsTests.cs

```diff
@@ -393,12 +400,38 @@ public static void Verify_InheritanceMethodResolution()
         var instance = new InheritanceDerived();
         Assert.Throws<MissingMethodException>(() => OnBase(instance));
```

Comment:
I didn't realize this before, but the static method case is also similar. In that in normal C# it's perfectly valid to call `InheritanceDerived.OnBase()`, while `UnsafeAccessor` will not allow it right now (across all runtimes).
I'm perfectly OK we ship "some" behavior which is consistent across the runtimes, but I'd ask for us to create an issue to track this going forward as there are apparently more cases (static methods, instance non-virtual methods, maybe something more...) which behave differently with `UnsafeAccessor` and direct call.

Reply:
I'd make this follow IL rules, not C# rules. C# can make weird things appear valid and then lower it into a different IL (C# will lower your example into a different IL than what you see in the source code).

Your comment made me look more into how MemberRef resolution works. It kept bugging me that UnsafeAccessors only do this for virtual methods. For MemberRef resolution, it is valid for any method to be moved to the base class, not just virtual, and the member will resolve there.

The current UnsafeAccessors approach is neither here nor there.

Another possible fix could be to use `IntroducedMethodIterator` on the CoreCLR side (that appears to have the right semantics, based on comments) and leave methods moving to base class for the future. It is odd to do this only for virtuals and it is definitely just an artifact of how MethodIterator works and not something we'd design as such.

I think if we add support for resolving in bases later, we could do this without causing a breaking change (will need to be careful about how we treat ambiguous matches), or we'd introduce an extra flag on the attribute to opt into resolving in bases in the worst case (if we come up with a scenario where it does introduce a break).